From 3d41c93556ece1dcadabde6c7c40f5691bb127e7 Mon Sep 17 00:00:00 2001
From: Piotr Tworek <tworaz666@gmail.com>
Date: Tue, 29 Sep 2015 08:18:00 +0200
Subject: [PATCH 11/15] Bug 1209446 - Make sure mFrameInProgress flag is set to
 true only when we actually begin drawing new frame. r=nical

--HG--
extra : rebase_source : ebb7d61be07e6cd9f2bcd920e3987dca9ac8c358

Signed-off-by: Raine Makelainen <raine.makelainen@jolla.com>
---
 gfx/layers/opengl/CompositorOGL.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/gfx/layers/opengl/CompositorOGL.cpp b/gfx/layers/opengl/CompositorOGL.cpp
index b6e6cef..b4b42cd 100644
--- a/gfx/layers/opengl/CompositorOGL.cpp
+++ b/gfx/layers/opengl/CompositorOGL.cpp
@@ -584,7 +584,6 @@ CompositorOGL::BeginFrame(const nsIntRegion& aInvalidRegion,
 
   MOZ_ASSERT(!mFrameInProgress, "frame still in progress (should have called EndFrame");
 
-  mFrameInProgress = true;
   gfx::Rect rect;
   if (mUseExternalSurfaceSize) {
     rect = gfx::Rect(0, 0, mSurfaceSize.width, mSurfaceSize.height);
@@ -606,6 +605,9 @@ CompositorOGL::BeginFrame(const nsIntRegion& aInvalidRegion,
     return;
   }
 
+  // We're about to actually draw a frame.
+  mFrameInProgress = true;
+
   // If the widget size changed, we have to force a MakeCurrent
   // to make sure that GL sees the updated widget size.
   if (mWidgetSize.width != width ||
-- 
2.1.4

